components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -11,23 +11,30 @@ import org.mozilla.geckoview.GeckoResponse
 import org.mozilla.geckoview.GeckoRuntime
 import org.mozilla.geckoview.GeckoSession
 import kotlinx.coroutines.experimental.launch
+import org.mozilla.geckoview.GeckoSessionSettings
 
 /**
  * Gecko-based EngineSession implementation.
  */
+@Suppress("TooManyFunctions")
 class GeckoEngineSession(
-    runtime: GeckoRuntime
+    private val runtime: GeckoRuntime,
+    privateMode: Boolean = false
```

Comment:
I think we should keep private browsing and tracking protection separated. Is this possible with GeckoView?

Reply:
OK, let's postpone private browsing. I will remove the commit.